### PR TITLE
Add sig verify to demo

### PR DESCRIPTION
**Pull request by justinbarry**

### CODE REVIEW

File: apps/demo-app/src/components/blocking-button.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        void handleClick();
+      }}
+      disabled={loading}
+    >
+      {loading ? (
+        <p>
+          <ArrowPathIcon className="h-5 w-5 animate-spin content-center" />
+        </p>
+      ) : (
+        text
```
Comment (BurntVal) [inline apps/demo-app/src/components/blocking-button.tsx:41]: Is this meant to be "text"

File: apps/demo-app/pages/api/check-signature.ts [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+export function verifyXionSignature(
+  address: string,
+  pubKey: string,
+  messageString: string,
+  signature: string,
+): boolean {
+  const signatureBuffer = Buffer.from(signature, "base64");
+  const uint8Signature = new Uint8Array(signatureBuffer); // Convert the buffer to an Uint8Array
+  const pubKeyValueBuffer = Buffer.from(pubKey, "base64"); // Decode the base64 encoded value
+  const pubKeyUint8Array = new Uint8Array(pubKeyValueBuffer); // Convert the buffer to an Uint8Array
```
Comment (BurntNerve) [inline apps/demo-app/pages/api/check-signature.ts:30]: Is there a situation where any of these parameters are empty/malformed/invalid? Do we need to do any checking inside the function before performing the changes?

**Review by BurntVal — APPROVED** [review]
LGTM

File: apps/demo-app/pages/api/check-signature.ts [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    granter,
+  });
+  url.search = params.toString();
+  const data = await fetch(url, {
+    cache: "no-store",
+  })
+    .then((response): Promise<QueryGrantsResponse> => response.json())
+    .catch((err) => {
+      console.error("Could not fetch grants info", err);
+    });
```
Comment (BurntNerve) [inline apps/demo-app/pages/api/check-signature.ts:99]: Seems like we're mixing async/await syntax and promise .then chaining. Not an immediate issue but was tough to parse at first.

File: apps/demo-app/pages/api/check-signature.ts [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+export async function verifyGrants(
+  granter: string,
+  grantee: string,
+): Promise<boolean> {
+  const baseUrl = `${testnetChainInfo.rest}/cosmos/authz/v1beta1/grants`;
+  const url = new URL(baseUrl);
+  const params = new URLSearchParams({
+    grantee,
+    granter,
+  });
```
Comment (BurntNerve) [inline apps/demo-app/pages/api/check-signature.ts:91]: Do we need to validate granter/grantee?

File: apps/demo-app/pages/api/check-signature.ts [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+      errors.push("itemId is required");
+    }
+
+    if (!message && typeof message === "string") {
+      errors.push("message is required");
+    }
+
+    if (!signature && typeof signature === "string") {
+      errors.push("signature is required");
+    }
```
Comment (BurntNerve) [inline apps/demo-app/pages/api/check-signature.ts:138]: Could maybe condense these and colocate the error messages so it's easier to tell what the issue is at a quick glance. Not an issue worth not approving, but a thought.

```typescript
const requiredParams = {
  userSessionAddress: "userSessionAddress is required",
  userSessionPubKey: "userSessionPubKey is required",
  metaAccountAddress: "metaAccountAddress is required",
  message: "message is required",
  signature: "signature is required",
};

for (const [param, errorMsg] of Object.entries(requiredParams)) {
  if (!query[param] || !isString(query[param])) {
    errors.push(errorMsg);
  }
}
```

**Review by BurntNerve — APPROVED** [review]
Overall looks really good! Just made a couple suggestions for condensing a few things.

File: apps/demo-app/pages/api/check-signature.ts (hunk 2 above)
Comment (justinbarry) [reply apps/demo-app/pages/api/check-signature.ts:30]: The verifyADR36Amino call should confirm these for us.

File: apps/demo-app/pages/api/check-signature.ts (hunk 5 above)
Comment (justinbarry) [reply apps/demo-app/pages/api/check-signature.ts:138]: That is clever!

Comment (justinbarry) [reply apps/demo-app/pages/api/check-signature.ts:138]: Sadly, typescript figure out the types like this correctly :(

File: apps/demo-app/pages/api/check-signature.ts (hunk 4 above)
Comment (justinbarry) [reply apps/demo-app/pages/api/check-signature.ts:91]: We do not, the chain will do when we go to verify.

File: apps/demo-app/src/components/blocking-button.tsx (hunk 1 above)
Comment (justinbarry) [reply apps/demo-app/src/components/blocking-button.tsx:41]: Haha, it's a param coming in from the parent component.

**Pull request merged**